// io-elements/lib/Modules/Avatar/Avatar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _reactNative = require("react-native");
var _react = _interopRequireDefault(require("react"));
var _colors = require("../../utils/colors");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var radiusDefaults = {
  full: 100,
  lg: 14,
  md: 12,
  sm: 8
};
var Sizes = {
  lg: 60,
  md: 30,
  sm: 20
};
var BorderSize = {
  lg: 3,
  md: 2,
  sm: 1
};
var Avatar = function Avatar(_ref) {
  var _ref$size = _ref.size,
    size = _ref$size === void 0 ? 80 : _ref$size,
    _ref$source = _ref.source,
    source = _ref$source === void 0 ? "" : _ref$source,
    _ref$raduis = _ref.raduis,
    raduis = _ref$raduis === void 0 ? "md" : _ref$raduis,
    _ref$borderSize = _ref.borderSize,
    borderSize = _ref$borderSize === void 0 ? "sm" : _ref$borderSize,
    textStyle = _ref.textStyle,
    _ref$text = _ref.text,
    text = _ref$text === void 0 ? "Avatar" : _ref$text;
  var Components = _colors.Colors.Components,
    Context = _colors.Colors.Context;
  var defaultRadius = raduis === "full" ? size : typeof raduis === "number" ? raduis : radiusDefaults[raduis] || radiusDefaults.md;
  var defaultSize = typeof size === "number" ? size : Sizes[size] || Sizes.md;
  var defaultBorderSize = typeof borderSize === "number" ? borderSize : BorderSize[borderSize] || BorderSize.md;
  return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
    style: [{
      width: defaultSize,
      height: defaultSize
    }, {
      borderRadius: typeof defaultRadius === "number" ? defaultRadius : 0,
      borderWidth: defaultBorderSize,
      borderColor: Components.defaultBorder
    }, {
      backgroundColor: Components.ZincBg,
      alignItems: "center",
      justifyContent: "center"
    }]
  }, source && /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, typeof source === "string" ? /*#__PURE__*/_react["default"].createElement(_reactNative.Image
  // @ts-ignore
  , {
    style: {
      width: "99%",
      height: "99%",
      borderRadius: typeof defaultRadius === "number" ? defaultRadius : 0
    },
    source: {
      uri: source
    }
  }) : /*#__PURE__*/_react["default"].createElement(_reactNative.Image
  // @ts-ignore
  , {
    style: {
      width: "99%",
      height: "99%",
      borderRadius: typeof defaultRadius === "number" ? defaultRadius : 0
    },
    source: source
  })), source === "" && /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
    style: [{
      fontSize: defaultSize * 0.25,
      color: Context.text,
      fontWeight: "200"
    }, textStyle]
  }, text));
};
var _default = Avatar;
exports["default"] = _default;
